applications/matter_bridge/src/bridged_devices_creator.cpp

```diff
@@ -0,0 +1,384 @@
+	}
+	return err;
+}
+
+#ifdef CONFIG_BRIDGED_DEVICE_SIMULATED
+BridgedDeviceDataProvider *CreateSimulatedProvider(int deviceType)
+{
+	return BridgeFactory::GetSimulatedDataProviderFactory().Create(
+		static_cast<MatterBridgedDevice::DeviceType>(deviceType), BridgeManager::HandleUpdate);
+}
+#endif /* CONFIG_BRIDGED_DEVICE_SIMULATED */
+
+#ifdef CONFIG_BRIDGED_DEVICE_BT
+
+struct BluetoothConnectionContext {
+	int deviceType;
+	char nodeLabel[MatterBridgedDevice::kNodeLabelSize];
+	BLEBridgedDeviceProvider *provider;
+	bt_addr_le_t address;
+	chip::Optional<uint8_t> index;
+	chip::Optional<uint16_t> endpointId;
+};
+
+void BluetoothDeviceConnected(bt_gatt_dm *discoveredData, bool discoverySucceeded, void *context)
+{
+	if (!context) {
+		return;
+	}
+
+	BluetoothConnectionContext *ctx = reinterpret_cast<BluetoothConnectionContext *>(context);
+	chip::Platform::UniquePtr<BluetoothConnectionContext> ctxPtr(ctx);
+
+	VerifyOrExit(discoveredData && discoverySucceeded, );
+	VerifyOrExit(ctxPtr->provider->ParseDiscoveredData(discoveredData) == 0, );
+
+	/* Schedule adding device to main thread to avoid overflowing the BT stack. */
+	chip::DeviceLayer::PlatformMgr().ScheduleWork(
+		[](intptr_t context) {
+			BluetoothConnectionContext *ctx = reinterpret_cast<BluetoothConnectionContext *>(context);
+			if (CHIP_NO_ERROR != AddMatterDevice(ctx->deviceType, ctx->nodeLabel, ctx->provider, ctx->index,
+							     ctx->endpointId)) {
+				BLEConnectivityManager::Instance().RemoveBLEProvider(ctx->address);
+			}
+		},
+		reinterpret_cast<intptr_t>(ctx));
+	return;
+
+exit:
+	BLEConnectivityManager::Instance().RemoveBLEProvider(ctx->address);
+}
+
+BridgedDeviceDataProvider *CreateBleProvider(int deviceType)
+{
+	return BridgeFactory::GetBleDataProviderFactory().Create(
+		static_cast<MatterBridgedDevice::DeviceType>(deviceType), BridgeManager::HandleUpdate);
+}
+#endif /* CONFIG_BRIDGED_DEVICE_BT */
+} /* namespace */
+
+#ifdef CONFIG_BRIDGED_DEVICE_BT
+CHIP_ERROR BridgedDeviceCreator::CreateDevice(int deviceType, const char *nodeLabel, bt_addr_le_t btAddress,
+					      chip::Optional<uint8_t> index, chip::Optional<uint16_t> endpointId)
+{
+	CHIP_ERROR err;
+
+	/* Check if there is already existing provider for given address.
+	 * In case it is, the provider was either connected or recovered before. All what needs to be done is creating
+	 * new Matter endpoint. In case there isn't, the provider has to be created.
+	 */
+	BLEBridgedDeviceProvider *provider = BLEConnectivityManager::Instance().FindBLEProvider(btAddress);
+	if (provider) {
+		return AddMatterDevice(deviceType, nodeLabel, provider, index, endpointId);
+	}
+
+	/* The EnvironmentalSensor is a dummy device type, so it is not saved in the persistent storage. For now the
+	 * only BLE service handling temperature and humidity device types is environmental sensor, so it's necessary to
+	 * perform mapping. */
+	if (deviceType == MatterBridgedDevice::DeviceType::TemperatureSensor ||
```

Comment:
Can we remove specific dependencies like this one from this generic file, or it is not possible at this moment?

Reply:
I would love to rework this module to not have it, but it was too much work to already very big PR, so I left it as temporary inconvenience. I would imagine we should not ask user to assign device type, but get it on our own based on e.g. GATT discovery results parsed by some new parser module that would map bluetooth types to Matter device types. We have a KRKNWK-17561 that could cover it.